app/models/vm_or_template/operations/configuration.rb

```diff
@@ -73,8 +73,8 @@ def disconnect_floppies
 
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
-    if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
+    if options[:datastore] || options[:datastore_id]
```

Comment:
@bronaghs, @abellotti  wouldn't it be considered as breaking the API ?

In [1] the datastore is exposed as `diskscsi##.datastore=<datastore name>`, therefore changing it to `datastore_name` will cause any existing users/customer scripts not to work.

I can add  `diskscsi##.datastore_name` in addition, with the same semantics as the  `diskscsi##.datastore`, and later on to deprecate the latter, but I'm not familiar with the API deprecation policy.

[1] https://access.redhat.com/documentation/en-US/Red_Hat_CloudForms/3.2/html/Red_Hat_CloudForms_SOAP_API/Defining_new_disks_during_provisioning.html